src/webgl/loading.js

```diff
@@ -178,6 +202,54 @@ p5.prototype.loadModel = function(path) {
   return model;
 };
 
+function parseMtl(mtlPath){ //accepts mtlPath to load file
+  return new Promise((resolve,reject)=>{
+    // console.log('parser is called');
+    let currentMaterial = null;
+    p5.prototype.loadStrings(
```

Comment:
I think rather than calling this on `p5.prototype`, which will have no `this`, you'd want to call `loadStrings` on a p5 instance. That could mean calling `await parseMtl(this, mtlPath);` inside `loadModel` and then call `loadStrings` on the first argument, although it might be a nicer separation of concerns to separate loading the strings and the parsing, so it becomes:
```js
this.loadStrings(
  mtlPath,
  (mtlLines) => {
    const materials = parseMtl(mtlLines);
    parseObj(model, lines, materials)
  },
  () => {
    // handle failure
  }
)
```
Another benefit of this is that you maybe don't need `parseMtl` to resolve or reject, since the `loadStrings` part handles the case where it can't fetch the file, so `parseMtl` can look more like `parseObj`.